core/client/fs/src/main/java/alluxio/client/file/FileSystemContext.java

```diff
@@ -201,6 +205,7 @@ private FileSystemContext(Subject subject, AlluxioConfiguration alluxioConf) {
   private FileSystemContext(ClientContext ctx) {
     Preconditions.checkNotNull(ctx, "ctx");
     mClientContext = ctx;
+    mPathIndexer = new PathIndexer(mClientContext.getConf());
```

Comment:
we should move the path index to the `InstancedConfiguration` instead of `FileSystemContext`